src/bundles/SellerRegistration/components/ProductsForm/ProductsForm.js

```diff
@@ -43,6 +47,14 @@ class ProductsForm extends BaseForm {
       removeProduct(model, id);
   }
 
+  onEmptyNav() {
+    const {navFromProducts} = this.props;
+    let servicesEmpty = !some(Object.values(this.props.services));
```

Comment:
Yes. Services that have been selected, then unselected, get flipped to false, not removed so need to check for trues.